### PR TITLE
DOC: how to write a penalty and a datafit

### LINKED ISSUE #31: DOC : add example on how to write a custom Penalty
A la https://benchopt.github.io/how.html#example

### PR DESCRIPTION
**Pull request by mathurinm**
closes #31 (also adds a BaseDatafit class and shows how to write a custom datafit)

### CODE REVIEW

**Comment by mathurinm** [comment]
https://292-306296806-gh.circle-artifacts.com/0/dev/add.html

File: andersoncd/penalties.py [hunk 1]
```diff
@@ -22,13 +23,13 @@ def prox_1d(self, value, stepsize, j):
...
     @abstractmethod
-    def subdiff_distance(self, w, neg_grad, ws):
+    def subdiff_distance(self, w, grad, ws):
         """Distance of gradient to subdifferential of penalty for feature j.
 
         w : array, shape (n_features,)
             Coefficient vector.
-        neg_grad: array, shape (n_features,)
-            Minus the value of the gradient of the datafit at w.
+        grad: array, shape (n_features,)
```
Comment (QB3) [inline andersoncd/penalties.py:31]: with the code of the L1 penalty, it seems that grad has the size of the working set

**Comment by mathurinm** [comment]
Merge ?

**Comment by QB3** [comment]
I merge when green !

**Pull request merged**